### PR TITLE
fix deprecation warning from setup-gcloud

### PR DESCRIPTION
**Pull request by dekkagaijin**
Fixes warnings e.g. https://github.com/sigstore/cosign/actions/runs/1555909393

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: .github/workflows/e2e_tests.yml [hunk 1]
```diff
@@ -32,11 +32,14 @@ jobs:
       - uses: actions/setup-go@v2
         with:
           go-version: '1.17.x'
+      - id: auth
+        uses: google-github-actions/auth@v0.4.2
+        with:
+          credentials_json: ${{ secrets.GCP_CI_SERVICE_ACCOUNT }}
```
Comment (mattmoor) [inline .github/workflows/e2e_tests.yml:38]: Can we file an issue to replace this with token federation?

**Review by mattmoor — APPROVED** [review]

File: .github/workflows/e2e_tests.yml (hunk 1 above)
Comment (dekkagaijin) [reply .github/workflows/e2e_tests.yml:38]: yes, but that requires additional setup. I can add an issue

**Comment by dekkagaijin** [comment]
sigh... looks like the setup action dropped support for base64 input

**Comment by bobcallaway** [comment]
I think this is fixed via workload identity in #1330

**Pull request closed without merging**